genie-client/src/main/java/com/netflix/genie/client/JobClient.java

```diff
@@ -476,13 +469,37 @@ public InputStream getJobStdout(final String jobId) throws IOException, GenieCli
      */
     public InputStream getJobStderr(
         final String jobId
+    ) throws IOException, GenieClientException {
+        return getJobOutputFile(jobId, "stderr");
+    }
+
+    /**
+     * Method to fetch an output file for a job from Genie.
+     *
+     * <p>
+     * <b>NOTE</b>: If the specified outputFilePath is a directory, then the directory
+     * manifest is returned.
+     * </p>
+     *
+     * @param jobId The id of the job whose output file is desired.
+     * @param outputFilePath The path to the file within output directory.
+     * @return An input stream to the output file contents.
+     * @throws GenieClientException If the response received is not 2xx.
+     * @throws IOException          For Network and other IO issues.
+     */
+    public InputStream getJobOutputFile(
+        final String jobId, final String outputFilePath
     ) throws IOException, GenieClientException {
         if (StringUtils.isEmpty(jobId)) {
             throw new IllegalArgumentException("Missing required parameter: jobId.");
         }
-        final ResponseBody body = this.jobService.getJobStderr(jobId).execute().body();
+        if (StringUtils.isEmpty(outputFilePath)) {
```

Comment:
@vickyuec I think this can be empty as it would return the root of the output directory. At least if you want to be complete.

Reply:
Fixed this, and updated IT to test it.